### PR TITLE
Update linter/formatter to target sub-directories

### PR DESCRIPTION
**Pull request by genedwards**
Updated to target sub-directories for the linter and formatter.

Followed some suggestions in this issue: https://github.com/prettier/prettier/issues/2078
I haven't verified if the Java files also have issues with subdirectories, but we'll tackle that separately.

Tested with commit [caaa4b7](https://github.com/googleinterns/step88-2020/pull/58/commits/caaa4b789bb0f9c7737cef4bbdb39b2ca41c261e)
Which passed after running `make pretty` and fixing/disabling lint errors. I've added TODOs for disabled lines.
image

Also tested deeply nested files locally which were picked up by the linter:
image

### CODE REVIEW

File: frontend/src/attraction/Attraction.js [hunk 1]
```diff
@@ -5,10 +5,11 @@ import styles from './Attraction.module.css';
  * Attraction contains the image of the attraction.
  * imageUrl contains source of image.
  */
-function Attraction({imageUrl}) {
+function Attraction({ imageUrl }) {
   return (
     <div className={styles.attractionContainer}>
-      <img src={imageUrl}/>
+      {/* TODO: Add meaningful alt text. */}
```
Comment (genedwards) [inline frontend/src/attraction/Attraction.js:11]: Added TODO

File: frontend/src/map/Map.js [hunk 2]
```diff
@@ -8,6 +8,8 @@ import styles from './Map.module.css';
  * @param {string} mode either 'pins' or 'directions' to put on the map
  * @param {Object} centerLocation the center of the map, the location of the attraction the user initially searched
  */
+// TODO: Remove temporarily disabled linter.
+// eslint-disable-next-line
```
Comment (genedwards) [inline frontend/src/map/Map.js:12]: Added TODO to remove the line which temporarily disabled the linter

File: frontend/src/map/Map.js [hunk 3]
```diff
@@ -31,6 +33,8 @@ function Map({ destinations, mode, centerLocation }) {
 
         for (const place of mockData) {
           const location = { lat: place.lat, lng: place.lng };
+          // TODO: Remove temporarily disabled linter.
+          // eslint-disable-next-line
```
Comment (genedwards) [inline frontend/src/map/Map.js:37]: Added TODO to remove the line which temporarily disabled the linter

File: makefile [hunk 4]
```diff
@@ -14,9 +14,9 @@ node_modules:
...
 pretty: node_modules
-	$(PRETTIER) --write frontend/src/*.css
-	$(PRETTIER) --write frontend/src/*.js
+	$(PRETTIER) --write 'frontend/src/*.css' 'frontend/src/**/*.css'
+	$(PRETTIER) --write 'frontend/src/*.js' 'frontend/src/**/*.js'
 	find backend/src/main -iname *.java | xargs $(CLANG_FORMAT) -i
 
 validate: node_modules
-	$(ESLINT) frontend/src/*.js
+	$(ESLINT) 'frontend/src/*.js' 'frontend/src/**/*.js'
```
Comment (genedwards) [inline makefile:22]: Followed some suggestions in this issue: prettier/prettier#2078

File: makefile [hunk 5]
```diff
@@ -14,9 +14,9 @@ node_modules:
 	npm install clang-format prettier eslint react@16.13.1 eslint-config-react-app babel-eslint eslint-plugin-import eslint-plugin-flowtype eslint-plugin-jsx-a11y eslint-plugin-react eslint-plugin-react-hooks
 
 pretty: node_modules
-	$(PRETTIER) --write frontend/src/*.css
-	$(PRETTIER) --write frontend/src/*.js
+	$(PRETTIER) --write 'frontend/src/*.css' 'frontend/src/**/*.css'
```
Comment (mecolmg) [inline makefile:17]: FYI, I didn't know about this before, but you can use [`npx`](https://www.npmjs.com/package/npx) which will install a binary package if it isn't already in the node_modules. Not worth changing, just FYI.

Comment (genedwards) [reply makefile:17]: oh awesome, I didn't know npx installed it as well
good to know

File: .github/workflows/lint.yml [hunk 6]
```diff
@@ -13,16 +13,16 @@ jobs:
       run: make node_modules
     - name: Check CSS Formatting
       if: always()
-      run: node_modules/prettier/bin-prettier.js -c frontend/src/*.css
+      run: node_modules/prettier/bin-prettier.js -c 'frontend/src/*.css' 'frontend/src/**/*.css'
```
[the remark this answers is not in the document]
Comment (genedwards) [reply .github/workflows/lint.yml:16]: tested locally with your suggestions and it picked up the frontend/src/App.js
updated

File: makefile (hunk 5 above)
[the remark this answers is not in the document]
Comment (genedwards) [reply makefile]: updated

**Pull request merged**